### PR TITLE
Extract infer-files and build-and-release-docker-image to scripts

### PR DESCRIPTION
**Pull request by fcmichael**

### CODE REVIEW

File: .github/workflows/cicd.yaml [hunk 1]
```diff
@@ -110,7 +100,7 @@ jobs:
           retention-days: 1
   releaseLibrary:
     name: 📦 Release NPM artifact
-    if: needs.checks.outputs.packageJsonFilesExists == 'true' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/hotfix-'))
+    # if: needs.checks.outputs.packageJsonFilesExists == 'true' && (github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/heads/hotfix-'))
```
Comment (kamil-perczynski) [inline .github/workflows/cicd.yaml:103]: 🔥🚒

File: actions/infer-files/infer-files.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#! /bin/bash
+DOCKERFILE_EXISTS=$([ -f Dockerfile ] && echo "true" || echo "false");
```
[the remark this answers is not in the document]
Comment (fcmichael) [reply actions/infer-files/infer-files.sh]: rewrited

File: .github/workflows/cicd.yaml (hunk 1 above)
Comment (fcmichael) [reply .github/workflows/cicd.yaml]: uncommented

File: actions/release-docker-image/action.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  IMAGE_TAG:
+    description: A deployed docker image tag
+    required: true
+runs:
+  using: "composite"
+  steps:
+    - shell: bash
+      run: chmod +x ${{ github.action_path }}/release-docker-image.sh
+    - shell: bash
+      run: ${{ github.action_path }}/release-docker-image.sh ${{ github.action_path }}
```
Comment (kamil-perczynski) [inline actions/release-docker-image/action.yml:19]: The script is not using the `${{ github.action_path }}` parameter

Comment (fcmichael) [reply actions/release-docker-image/action.yml]: done

**Review by kamil-perczynski — APPROVED** [review]

**Pull request merged**